src/test/java/org/apache/commons/crypto/stream/AbstractCipherStreamTest.java

```diff
@@ -197,46 +224,246 @@ protected void doByteBufferRead(final String cipherClass, final boolean withChan
+        } catch (IOException ex) {
+            Assert.assertTrue(ex.getMessage().equals("Stream closed"));
+        } 
+
+        try { // Test closing a closed stream.
+            out.close(); // Don't throw exception.
+        } catch (IOException ex) {
+            Assert.fail("Should not throw exception closing a closed stream.");
+        } 
+        
+        try { // Test checkStreamCipher
+            CryptoInputStream.checkStreamCipher(getCipher(cipherClass));
+        } catch (IOException ex) {
+            Assert.assertTrue(ex.getMessage().equals("AES/CTR/NoPadding is required"));
+        } finally {
+            in.close();
+        }
 
-        out.flush();
+        try { // Test unsupported operation handling.
+            in = getCryptoInputStream(new ByteArrayInputStream(encData), 
+                    getCipher(cipherClass), defaultBufferSize, iv, false);
+            in.mark(0); // Should not throw an exception.
+            assertEquals(false, in.markSupported());
+            in.reset();
+            Assert.fail("Expected IOException.");
+        } catch (IOException ex) {
+            Assert.assertTrue(ex.getMessage().equals("Mark/reset not supported"));
+        }  
+    }
 
-        try (InputStream in = getCryptoInputStream(
-                new ByteArrayInputStream(encData), getCipher(cipherClass),
-                defaultBufferSize, iv, withChannel)) {
-            buf = ByteBuffer.allocate(dataLen + 100);
-            byteBufferReadCheck(in, buf, 0);
+    protected void doFieldGetterTest(final String cipherClass, ByteArrayOutputStream baos,
+            final boolean withChannel) throws Exception {
+        if (AbstractCipherTest.OPENSSL_CIPHER_CLASSNAME.equals(cipherClass)) {
+            if (!Crypto.isNativeCodeLoaded()) {
+                return; // Skip this test if no JNI
+            }
         }
-    }
+        
+        CryptoInputStream in = getCryptoInputStream(
+                new ByteArrayInputStream(encData), getCipher(cipherClass), defaultBufferSize, 
+                iv, withChannel); 
+
+        Properties props = new Properties();
+        String bufferSize = Integer.toString(defaultBufferSize / 2);
+        props.put(CryptoInputStream.STREAM_BUFFER_SIZE_KEY, bufferSize);
+
+        Assert.assertEquals(CryptoInputStream.getBufferSize(props), Integer.parseInt(bufferSize));
+        Assert.assertEquals(in.getBufferSize(), defaultBufferSize);
+        Assert.assertEquals(in.getCipher().getClass(), Class.forName(cipherClass));
+        Assert.assertEquals(in.getKey().getAlgorithm(), "AES");
+        Assert.assertEquals(in.getParams().getClass(), IvParameterSpec.class);
+        Assert.assertNotNull(in.getInput());
+
+        CryptoOutputStream out = getCryptoOutputStream(baos, getCipher(cipherClass), 
+                defaultBufferSize, iv, withChannel);
 
+        Assert.assertEquals(out.getOutBuffer().capacity(), defaultBufferSize + 16);
```

Comment:
Is there a constant for the `16` anywhere?

Reply:
I'll replace it with getBlockSize.